include/ppx/fs.h

```diff
@@ -46,27 +46,69 @@ namespace ppx::fs {
 void set_android_context(android_app* androidContext);
 #endif
 
-//! @class File
-//!
-//!
+// Abstract a static, regular file on all platforms.
+// This class doesn't handle sockets, or file which content is not constant
+// for the lifetime of this class.
 class File
 {
+    // Internal enum to know which handle is currently valid.
+    enum FileHandleType
+    {
+        // Default handle, no file associated.
+        BAD_HANDLE = 0,
+        // The file is accessible throught a stream.
+        STREAM_HANDLE = 1,
+        // The file is accessible through an Android asset handle.
+        ASSET_HANDLE = 3
+    };
+
 public:
-    bool   Open(const std::filesystem::path& path);
-    bool   IsOpen() const;
-    size_t Read(void* buf, size_t count);
-    size_t GetLength();
-    void   Close();
-#if defined(PPX_ANDROID)
-    const void* GetBuffer();
-#endif
+    File();
+    File(const File& other)  = delete;
+    File(const File&& other) = delete;
+    ~File();
+
+    // Open a file given a specific path.
+    // path: the path of the file to open.
+    //  - On desktop, loads the regular file at `path`. (memory-mapping availability is implementation defined).
+    //  - On Android, relative path are assumed to be loaded from the APK, those are memory mapped.
+    //                absolute path are loaded as regular files (mapping availability is implementation defined).
+    //
+    // - This API only supports regular files.
+    // - This API expects the file not to change size or content while this handle is open.
+    // - This class supports RAII. File will be closed on destroy.
+    bool Open(const std::filesystem::path& path);
+
+    // Read `size` bytes from the file into `buffer`.
+    // buffer: a pointer to a buffer with at least `count` writable bytes.
+    // count: the maximum number of bytes to write to `buffer`.
+    // Returns the number of bytes written to `buffer`.
+    //
+    // - The file has an internal cursor, meaning the next read will start at the end of the last read.
+    // - If the file size is larger than `count`, the read stops at `count` bytes.
+    size_t Read(void* buffer, size_t count);
+
+    // Is this file readable?
+    bool IsValid() const;
+    // Is this file mapped in memory? See `File::GetPointer()`.
+    bool IsMapped() const;
+
+    // Gets the total size in bytes of the file from the start.
+    size_t GetLength() const;
+    // Gets a readable pointer to a beginning of the file. Behavior undefined is `File::IsMapped()` is false.
+    const void* GetPointer() const;
```

Comment:
Personally I think GetDataIfMapped would make it more readable, but up to you

Reply:
Renamed to `GetMappedData()`. The `IfMapped` parts seems like I'd be allowed to call this function to check if the pointer is valid. While the file-mapping is not implemented on all platforms, I'd prefer to split `IsMapped()` and `GetMappedData()` logics.
(Open to refactorize that in a single (GetMappedData()) and check for null if no platform gives us a constraint making this harder to implement.